### PR TITLE
[COM-111][COM-134][COM-135][COM-136] Texture Loading, Asset Manger, Material Class

### PR DESCRIPTION
**Pull request by Tissure**

### CODE REVIEW

File: Rendering/include/Data/VertexBufferLayout.h [hunk 1]
```diff
@@ -35,6 +36,7 @@ struct VertexBufferElement
         case GL_FLOAT:			return 4;
```
Comment (le-rich) [inline Rendering/include/Data/VertexBufferLayout.h:36]: Triggered by this spacing

File: Rendering/include/Resources/Primitives.h [hunk 2]
```diff
@@ -41,45 +39,6 @@ struct Shape
...
-            Vertex(glm::vec3(-0.5f,  0.5f, 0.0f))  // 3
-        };
-        indices =
-        {
-            0, 1, 2,
-            2, 3, 0
-        };
-    }
-};
-
```
Comment (le-rich) [inline Rendering/include/Resources/Primitives.h:80]: Removed becasue it's 2D?

**Review by le-rich — APPROVED** [review]

File: Rendering/include/Resources/Primitives.h (hunk 2 above)
Comment (Tissure) [reply Rendering/include/Resources/Primitives.h:80]: Bloat and unlikely to be used as is. Maybe as billboards but we'd have a billboard class.

**Review by imnorookie — APPROVED** [review]
looks good to me.

**Pull request merged**